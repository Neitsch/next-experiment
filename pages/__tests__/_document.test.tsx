import { shallow } from "enzyme";
import * as React from "react";

class Helper extends React.Component {
  public render() {
    return <div>Hi!</div>;
  }
}

describe("Documents", () => {
  it("renders", () => {
    jest.mock("styled-jsx/server", () => () => false);
    const Document = require("../_document").default;
    const rendered = shallow(
      <Document
        pageContext={{ theme: { palette: { primary: { main: "test" } } } }}
      />,
    );
    expect(rendered).toMatchSnapshot();
  });
  it("init props server", async () => {
    const Document = require("../_document").default;
    const renderPage = jest.fn().mockImplementation(fn => {
      const wrapped = fn(Helper);
      const component = wrapped({ pageContext: { sheetsRegistry: {} } });
      expect(shallow(component)).toMatchSnapshot();
    });
    const propsInit = await Document.getInitialProps({
      renderPage,
      req: {
        params: { nonce: "ghi" },
      },
    });
    const jssFun = renderPage.mock.calls[0][0];
    const JssComp = jssFun(Helper);
    const JssInst = <JssComp />;
    const JssRender = shallow(JssInst);

    expect(JssRender).toMatchSnapshot();
    expect(propsInit).toMatchSnapshot();
  });
  it("init props client", async () => {
    process.browser = true;
    const Document = require("../_document").default;
    const renderPage = jest.fn().mockImplementation(fn => {
      const wrapped = fn(Helper);
      const component = wrapped({
        pageContext: { sheetsRegistry: {} },
      });
      expect(shallow(component)).toMatchSnapshot();
    });
    const propsInit = await Document.getInitialProps({
      renderPage,
      req: {
        params: { nonce: "ghi" },
      },
    });
    const jssFun = renderPage.mock.calls[0][0];
    const JssComp = jssFun(Helper);
    const JssInst = <JssComp />;
    const JssRender = shallow(JssInst);

    expect(JssRender).toMatchSnapshot();
    expect(propsInit).toMatchSnapshot();

    const propsInit2 = await Document.getInitialProps({
      renderPage,
      req: {
        params: { nonce: "ghi" },
      },
    });
  });
});
