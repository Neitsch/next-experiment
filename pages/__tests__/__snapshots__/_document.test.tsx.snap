// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documents init props client 1`] = `
<div>
  Hi!
</div>
`;

exports[`Documents init props client 2`] = `<Helper />`;

exports[`Documents init props client 3`] = `
Object {
  "nonce": "ghi",
  "pageContext": Object {
    "sheetsRegistry": Object {},
  },
  "styles": <React.Fragment>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "[object Object]",
        }
      }
      id="jss-server-side"
      nonce="ghi"
    />
  </React.Fragment>,
}
`;

exports[`Documents init props client 4`] = `
<div>
  Hi!
</div>
`;

exports[`Documents init props server 1`] = `
<div>
  Hi!
</div>
`;

exports[`Documents init props server 2`] = `<Helper />`;

exports[`Documents init props server 3`] = `
Object {
  "nonce": "ghi",
  "pageContext": Object {
    "sheetsRegistry": Object {},
  },
  "styles": <React.Fragment>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "[object Object]",
        }
      }
      id="jss-server-side"
      nonce="ghi"
    />
  </React.Fragment>,
}
`;

exports[`Documents renders 1`] = `
<html
  dir="ltr"
  lang="en"
>
  <Head>
    <title>
      Condor Club
    </title>
    <meta
      charSet="utf-8"
    />
    <meta
      content="user-scalable=0, initial-scale=1, minimum-scale=1, width=device-width, height=device-height"
      name="viewport"
    />
    <meta
      property="csp-nonce"
    />
    <meta
      content="test"
      name="theme-color"
    />
  </Head>
  <body>
    <Main />
    <NextScript />
  </body>
</html>
`;
